### ISSUE TITLE
Setup an SSO service for admin interfaces

### ISSUE THREAD

**Issue by delroth** [NixOS/infra#375]
Some of our admin interfaces are currently stuck on Wireguard because they don't support (or badly support) HTTP authentication / authorization. Most notably right now: Alertmanager.

We should setup some kind of auth proxy - my first pick would be Authentik, but no strong preference. Then we can move some of the services behind it, and possibly also replace some stuff that currently have their own user accounts (Grafana? maybe even Hydra?).

**Comment by mweinelt** [comment]
My first pick would be kanidm. Our packaging in nixpkgs is great, and we have a great relationship with the upstream.

**Comment by delroth** [comment]
Sounds even better then!

**Comment by mweinelt** [comment]
[kanidm](https://search.nixos.org/options?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=kanidm) + [oauth2_proxy](https://search.nixos.org/options?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=oauth2_proxy) sounds like the right pairing for the described use case.

**Comment by mweinelt** [comment]
The alternative would be a simpler setup that binds to GitHub accounts, like https://dexidp.io/docs/connectors/github/, in which case we would use GitHub as an identity provider.

**Comment by jfly** [comment]
For the record, I believe @Erethon is working on something adjacent to this for [the Grist trial](https://www.getgrist.com/).

**Comment by Erethon** [comment]
> For the record, I believe [@Erethon](https://github.com/Erethon) is working on something adjacent to this for [the Grist trial](https://www.getgrist.com/).

Indeed, it's adjacent. One of the requirements for Grist is to have an OIDC provider, as such I'm looking into Dex which was already mentioned in this thread. If Dex sounds like a solution we would like to pursue (regardless of Grist), I can open a draft PR to test it out.

**Comment by mweinelt** [comment]
For alertmanager we just went with oauth2proxy against the GitHub team.
